File: src/python/test/test_blocking.py
import nifty

margin = [10,10,10]
blocking = nifty.tools.blocking(roiBegin=(0,0,0), roiEnd=(100,100,100), blockShape=(10,10,10))



print(blocking.roiBegin)
print(blocking.roiEnd)
print(blocking.blockShape)
print(blocking.blockShift)

blockWithHalo = blocking.getBlockWithHalo(0, margin)
block = blocking.getBlock(0)

outerBlock = blockWithHalo.outerBlock
innerBlock = blockWithHalo.innerBlock
innerBlockLocal = blockWithHalo.innerBlockLocal




print("B ",block.begin, block.end)
print("O ",outerBlock.begin, outerBlock.end)
print("I ",innerBlock.begin, innerBlock.end)
print("IL",innerBlockLocal.begin, innerBlockLocal.end)

File: src/test/test_blocking.cxx
#include "nifty/tools/blocking.hxx"



void blockingTest()
{   

    typedef nifty::tools::Blocking<3> Blocking;
    typedef typename Blocking::VectorType VectorType;

    VectorType roiBegin{0,0,0}, roiEnd{100,100,100}, blockShape{10,10,10}, blockShift{0,0,0}, halo{10,10,10};

    Blocking blocking(roiBegin, roiEnd, blockShape, blockShift);


    auto blockWithHalo = blocking.getBlockWithHalo(0, halo);

    const auto & innerBlock = blockWithHalo.innerBlock();
    const auto & outerBlock = blockWithHalo.outerBlock();

    NIFTY_TEST_OP(outerBlock.end()[0],==,20);
    NIFTY_TEST_OP(innerBlock.end()[0],==,10);

}

int main() {
	blockingTest();
}